src/com/facebook/buck/ide/intellij/IjProjectCommandHelper.java

```diff
@@ -233,6 +252,127 @@ private TargetGraph getProjectGraphForIde(
         buckEventBus, cell, enableParserProfiling, executor, passedInTargets);
   }
 
+  /**
+   * Instead of specifying the location of libraries in project files, Go requires libraries to be
+   * in locations consistent with their package name, either relative to GOPATH environment variable
+   * or to the "vendor" folder of a project. This method identifies code generation targets, builds
+   * them, and copy the generated code from buck-out to vendor, so that they are accessible by IDEs.
+   */
+  private ExitCode initGoWorkspace(TargetGraphAndTargets targetGraphAndTargets)
+      throws IOException, InterruptedException {
+    Map<BuildTargetSourcePath, Path> generatedPackages =
+        findCodeGenerationTargets(targetGraphAndTargets);
+    if (generatedPackages.size() == 0) {
+      return ExitCode.SUCCESS;
+    }
+    // Run code generation targets
+    ExitCode exitCode =
+        runBuild(
+            generatedPackages
+                .keySet()
+                .stream()
+                .map(BuildTargetSourcePath::getTarget)
+                .collect(ImmutableSet.toImmutableSet()));
+    if (exitCode != ExitCode.SUCCESS) {
+      return exitCode;
+    }
+
+    copyGeneratedGoCode(targetGraphAndTargets, generatedPackages);
+    return ExitCode.SUCCESS;
+  }
+
+  /**
+   * Assuming GOPATH is set to a directory higher or equal to buck root, copy generated code to the
+   * package path relative to the highest level vendor directory. Not handling the case of GOPATH
+   * lower than buck root for now, as it requires walking the directory structure, which can be
+   * expensive and unreliable (e.g., what if there are multiple src directory?).
+   */
+  private void copyGeneratedGoCode(
+      TargetGraphAndTargets targetGraphAndTargets,
+      Map<BuildTargetSourcePath, Path> generatedPackages)
+      throws IOException {
+    Path vendorPath;
+    ProjectFilesystem fs = cell.getFilesystem();
+    if (fs.exists(Paths.get("src"))) {
+      vendorPath = Paths.get("src", "vendor");
```

Comment:
What's vendor?

Reply:
`vendor` is a place to put third-party libraries. Go toolchain and IDEs is able to automatically pick up the Go packages in vendor directory. It is a good place to put Go packages that we don't want to put into source control.

Reply:
We do not have repository-specific details in Buck logic. You need to change it to smarter about detecting this location.

Reply:
It's not really repository-specific. Vendor folder is a golang concept: https://golang.org/cmd/go/#hdr-Vendor_Directories

Reply:
We don't really want to change Buck when there is a change in where these libs are located. You can put it in a configuration option and set it to whatever you need in your repository.

Reply:
Go toolchain and IDEs doesn't take configurations like that, unfortunately. Only packages in the main src directory and `vendor` directories are recognized.

Reply:
> Go toolchain and IDEs don't take configurations like that

But what if they change location? You would need to change this code to make it working, then deal with support of the different versions of these tools.

A configuration option makes it much easier to maintain. You can even use a configuration option with default value of `src/vendor` when it's not present.

Reply:
The approach here is very flexible, not repository specific at all. If a project has its own vendor folder at `GOPATH/src/foo/bar/vendor`, and the buck root is at `GOPATH`, then the generated code will be copied into `GOPATH/src/vendor`. If the buck root is at GOPATH/src/foo, the the generated code will be copied to `GOPATH/src/foo/vendor`. All of these vendor folders are recognized by Go toolchain and IDEs at the same time.

Of course, if you want the vendor folder location to be further configurable by users, I can do that too.

Overall, I see this PR a temporary solution to unblock Go developers using Buck. For long term solution, there are discussions on Bazel side for over a year ([1](https://github.com/bazelbuild/rules_go/issues/512), [2](https://github.com/bazelbuild/rules_go/issues/393)). We may want to watch with Bazel community and see what they end up with. Whatever the long term solution ends up, I am happy to assist the migration.

If you have further concerns about this feature, I can also add a configuration that disable this feature by default.

I also notice that @mkaczanowski recently commit [changes](https://github.com/facebook/buck/commit/9fe8c34606e1ee8dc0daa6e558434ec01608dab9) trying to support IDEs (great thanks to @mkaczanowski), but the performance is somehow quite bad (I need to dig a bit deeper to understand why), and I am not quite sure about symlink tree too, as people in Bazel run into problems with symlink trees ([3](https://github.com/bazelbuild/rules_go/issues/393#issuecomment-320710886))